mucommander-core/src/main/java/com/mucommander/ui/action/impl/ShowTerminalAction.java

```diff
@@ -129,12 +142,12 @@ private void revertToTableView() {
         LOGGER.info("Going to hide Terminal...");
         if (terminal != null) {
             mainFrame.getMainPanel().remove(terminal);
-            terminal.stop();
         }
         mainFrame.getSplitPane().setVisible(true);
         SwingUtilities.invokeLater(() -> {
-            // TODO focus on active terminal....
-            mainFrame.getSplitPane().requestFocusInWindow();
+            // TODO - it seems like keyboard buffer is being tainted or focus is not complete/exact......
+            mainFrame.getActivePanel().requestFocusInWindow();
+            mainFrame.getActiveTable().requestFocusInWindow();
```

Comment:
I think the last line should be enough

Reply:
fix pushed, plus avoid changing CWD if the same as previous one. Looking for a way to change CWD w/o using `cd` (or at least try to find out how MC does it so they don't print `cd` in console)